Bug 1879140: Set proper loglevel for auth errors

jhadvig opened the pull request:
Print the authentication failures that indicate that the user is not logged in on a `Debug` loglevel
```
  2020-07-01T17:39:31.533556276Z 2020/07/1 17:39:31 server: authentication failed: http: named cookie not present
  2020-07-01T17:39:34.888946631Z 2020/07/1 17:39:34 server: authentication failed: unauthenticated, no value for cookie openshift-session-token
```

Also updated the `klog` methods upon calling `redirectAuthError()` 

/assign @spadgett

spadgett commented on pkg/auth/auth.go:
```diff
@@ -369,21 +369,21 @@ func (a *Authenticator) CallbackFunc(fn func(loginInfo LoginJSON, successURL str
 
 		if urlState != cookieState.Value {
-			klog.Error("State in url does not match State cookie")
+			klog.Error("state in url does not match State cookie")
```
This one we should log at a higher level. I don't think we ever expect to see this.

jhadvig replied:
> I don't think we ever expect to see this.

in that case there should not be any log level set, since its not expected, right?

spadgett commented on pkg/auth/auth.go:
```diff
@@ -369,21 +369,21 @@ func (a *Authenticator) CallbackFunc(fn func(loginInfo LoginJSON, successURL str
 		token, err := oauthConfig.Exchange(ctx, code)
 		if err != nil {
-			klog.Infof("unable to verify auth code with issuer: %v", err)
+			klog.Errorf("unable to verify auth code with issuer: %v", err)
```
This one is not expected as well.

jhadvig replied:
then it should be logged by default, right?

spadgett commented on pkg/server/middleware.go:
```diff
@@ -43,13 +43,13 @@ func authMiddlewareWithUser(a *auth.Authenticator, handlerFunc func(user *auth.U
 		if !safe {
 			if err := a.VerifySourceOrigin(r); err != nil {
-				klog.Infof("invalid source origin: %v", err)
+				klog.V(4).Infof("invalid source origin: %v", err)
```
Also not expected. We should surface this one.

jhadvig replied:
if this one is expected, then shouldn't we put it on higher log level ?

spadgett commented on pkg/server/middleware.go:
```diff
@@ -43,13 +43,13 @@ func authMiddlewareWithUser(a *auth.Authenticator, handlerFunc func(user *auth.U
 
 			if err := a.VerifyCSRFToken(r); err != nil {
-				klog.Infof("invalid CSRFToken: %v", err)
+				klog.V(4).Infof("invalid CSRFToken: %v", err)
```
Also not expected.

spadgett reviewed (commented):
Some of these errors are unexpected and indicate a real problem. I think we should leave those logged by default.

jhadvig commented:
@spadgett comments addressed. PTAL

jhadvig commented:
/test e2e-gcp-console

spadgett commented on pkg/server/middleware.go:
```diff
@@ -25,7 +25,7 @@ func authMiddlewareWithUser(a *auth.Authenticator, handlerFunc func(user *auth.U
 		user, err := a.Authenticate(r)
 		if err != nil {
-			klog.Infof("authentication failed: %v", err)
+			klog.V(4).Infof("authentication failed: %v", err)
```
Will this get logged by default? We want to avoid the named cookie not present errors in the log, which is a red herring.

jhadvig replied:
So the default log level `Normal` which is represented by `2`, so no it should not be logged by default. Only if the console logLevel is set to `Debug`, which is represented by `4`.

spadgett replied:
You have everything at 4, but some of these we want to see by default. (See my initial review.)

spadgett replied:
Ping @jhadvig, this comment is still pending :)

jhadvig commented:
/retest

jhadvig commented:
/test e2e-gcp-console

jhadvig commented:
@spadgett I've addressed your comments. Although not both of these two errors will be surfaced only on higher loglevel:
```
2020-07-01T17:39:31.533556276Z 2020/07/1 17:39:31 server: authentication failed: http: named cookie not present
2020-07-01T17:39:34.888946631Z 2020/07/1 17:39:34 server: authentication failed: unauthenticated, no value for cookie openshift-session-token
```

since they are both executed upon [authentication](https://github.com/openshift/console/blob/master/pkg/auth/auth_openshift.go#L175-L181). If we would want to differentiate between them we would need to to some additional rafactoring.

jhadvig commented:
/test e2e-gcp-console

spadgett reviewed (approved):
/lgtm

jhadvig commented:
/test e2e-gcp-console

jhadvig commented:
/test e2e-gcp-console
